Mathlib/MeasureTheory/Measure/EverywherePos.lean

```diff
@@ -0,0 +1,296 @@
+measure subsets, but this does not say they also intersect `μ.everywherePosSubset s` along positive
+measure subsets.-/
+lemma isEverywherePos_everywherePosSubset_of_measure_ne_top
+    [OpensMeasurableSpace α] [InnerRegularCompactLTTop μ] (hs : MeasurableSet s) (h's : μ s ≠ ∞) :
+    μ.IsEverywherePos (μ.everywherePosSubset s) := by
+  intro x hx n hn
+  rcases mem_nhdsWithin_iff_exists_mem_nhds_inter.1 hn with ⟨u, u_mem, hu⟩
+  have A : 0 < μ (u ∩ s) := by
+    have : u ∩ s ∈ 𝓝[s] x := by rw [inter_comm]; exact inter_mem_nhdsWithin s u_mem
+    exact hx.2 _ this
+  have B : (u ∩ μ.everywherePosSubset s : Set α) =ᵐ[μ] (u ∩ s : Set α) :=
+    ae_eq_set_inter (ae_eq_refl _) (everywherePosSubset_ae_eq_of_measure_ne_top hs h's)
+  rw [← B.measure_eq] at A
+  exact A.trans_le (measure_mono hu)
+
+lemma IsEverywherePos.smul_measure (hs : IsEverywherePos μ s) {c : ℝ≥0∞} (hc : c ≠ 0) :
+    IsEverywherePos (c • μ) s :=
+  fun x hx n hn ↦ by simpa [hc.bot_lt, hs x hx n hn] using hc.bot_lt
+
+lemma IsEverywherePos.smul_measure_nnreal (hs : IsEverywherePos μ s) {c : ℝ≥0} (hc : c ≠ 0) :
+    IsEverywherePos (c • μ) s :=
+  hs.smul_measure (by simpa using hc)
+
+/-- If two measures coincide locally, then a set which is everywhere positive for the former is
+also everywhere positive for the latter. -/
+lemma IsEverywherePos.of_forall_exists_nhds_eq (hs : IsEverywherePos μ s)
+    (h : ∀ x ∈ s, ∃ t ∈ 𝓝 x, ∀ u ⊆ t, ν u = μ u) : IsEverywherePos ν s := by
+  intro x hx n hn
+  rcases h x hx with ⟨t, t_mem, ht⟩
+  apply lt_of_lt_of_le _ (measure_mono (inter_subset_left n t))
+  rw [ht _ (inter_subset_right n t)]
+  exact hs x hx _ (inter_mem hn (mem_nhdsWithin_of_mem_nhds t_mem))
+
+/-- If two measures coincide locally, then a set is everywhere positive for the former iff it is
+everywhere positive for the latter. -/
+lemma isEverywherePos_iff_of_forall_exists_nhds_eq (h : ∀ x ∈ s, ∃ t ∈ 𝓝 x, ∀ u ⊆ t, ν u = μ u) :
+    IsEverywherePos ν s ↔ IsEverywherePos μ s := by
+  refine ⟨fun H ↦ H.of_forall_exists_nhds_eq ?_, fun H ↦ H.of_forall_exists_nhds_eq h⟩
+  intro x hx
+  rcases h x hx with ⟨t, ht, h't⟩
+  exact ⟨t, ht, fun u hu ↦ (h't u hu).symm⟩
+
+/-- An open set is everywhere positive for a measure which is positive on open sets. -/
+lemma _root_.IsOpen.isEverywherePos [IsOpenPosMeasure μ] (hs : IsOpen s) : IsEverywherePos μ s := by
+  intro x xs n hn
+  rcases mem_nhdsWithin.1 hn with ⟨u, u_open, xu, hu⟩
+  apply lt_of_lt_of_le _ (measure_mono hu)
+  exact (u_open.inter hs).measure_pos μ ⟨x, ⟨xu, xs⟩⟩
+
+section TopologicalGroup
+
+variable {G : Type*} [Group G] [TopologicalSpace G] [TopologicalGroup G]
+  [LocallyCompactSpace G] [MeasurableSpace G] [BorelSpace G] {μ : Measure G}
+  [IsMulLeftInvariant μ] [IsFiniteMeasureOnCompacts μ] [InnerRegularCompactLTTop μ]
+
+open Pointwise
+
+/-- If a compact closed set is everywhere positive with respect to a left-invariant measure on a
```

Comment:
Probably, this does not require that the mesure is left invariant, but much less : for example, if translating the measure multiplies it by a >0 function…

Reply:
Indeed, the proof just requires that the measure is quasi-invariant by left-multiplication (i.e., left-multiplication respects zero measure sets), which by the way only depends on the measure class and not the measure itself. If I'm not mistaken, in each measure class which is quasi-invariant by left multiplication, there is a measure which is exactly invariant (maybe one needs some sigma-finiteness here to apply Radon-Nikodym), so the current statement does not really lose much generality -- we'll generalize if/when we need it!